File: DSA Unit 05/Lec 30/Q1.cpp
Single File Programming Question
Problem Statement

Rohan is arranging different box numbers using a Max Heap so that the largest box always stays on top.
  He inserts each number one by one into the heap and wants to know which box becomes the root after all insertions.

After inserting all the elements in the Max Heap, help him find and print the root.

Input format :
The first line of input consists of an integer n, representing the number of box labels.

The second line contains n space-separated integers, representing the box labels to be inserted into the Max Heap.

Output format :
The output prints an integer representing the root element of the Max Heap.

Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

1 ≤ n ≤ 100

1 ≤ box labels ≤ 100

Sample test cases :
Input 1 :
6
5 12 3 20 8 15
Output 1 :
20
Input 2 :
5
4 7 2 9 1
Output 2 :
9


#include <iostream>
using namespace std;

int main() {
    int n;
    cin >> n;
    int maxVal = -1;
    for (int i = 0; i < n; i++) {
        int x;
        cin >> x;
        if (x > maxVal) {
            maxVal = x;
        }
    }
    cout << maxVal;
    return 0;
}

File: Java/Unit 02/Lec 11/Q1.java
Single File Programming Question
Problem Statement



David is trying to solve a linear equation to find the values of x and y based on given coefficients a, b and c. Implement a class named Equation where the coefficients are inputted through a method named inputCoefficients(). Create an object of the Equation class, use the method to input the coefficients, and another method solveEquation() to solve and display the values of x and y.



Can you assist David in creating the program?



Formula: 

x = (c - b) / a

y = (c - a) / b

Input format :
The input consists of three double values :

The first line contains the input a representing the coefficient of the linear equations which is of type double.
The second line contains the input b representing the coefficient of the linear equations which is of type double.
The third line contains the input c representing the coefficient of the linear equations which is of type double.
Output format :
The first line of output prints one of the following:

If there is a solution for x, print "x = " followed by the result as a double value, rounded off to one decimal place.
If a is zero and no solution exists, print "a is zero".
The second line prints one of the following:

If there is a solution for y, print "y = " followed by the result as a double value, rounded off to one decimal place.
If b is zero and no solution exists, print "b is zero".


Refer to the sample output for formatting specifications.

Code constraints :
The given test cases fall under the following constraints:

0.0 ≤ a, b, c ≤ 100.0

Sample test cases :
Input 1 :
2.0
3.0
7.0
Output 1 :
x = 2.0
y = 1.7
Input 2 :
100.0
0.0
50.0
Output 2 :
x = 0.5
b is zero
Input 3 :
0.0
3.0
7.0
Output 3 :
a is zero
y = 2.3
Input 4 :
1.5
6.2
1.5
Output 4 :
x = -3.1
y = 0.0





  import java.util.Scanner;

class Equation {
    double a, b, c;

    void inputCoefficients() {
        Scanner sc = new Scanner(System.in);
        a = sc.nextDouble();
        b = sc.nextDouble();
        c = sc.nextDouble();
    }

    void solveEquation() {
        if (a == 0) {
            System.out.println("a is zero");
        } else {
            double x = (c - b) / a;
            System.out.printf("x = %.1f\n", x);
        }

        if (b == 0) {
            System.out.println("b is zero");
        } else {
            double y = (c - a) / b;
            System.out.printf("y = %.1f\n", y);
        }
    }
}

public class Main {
    public static void main(String[] args) {
        Equation obj = new Equation();
        obj.inputCoefficients();
        obj.solveEquation();
    }
}
